scripts/globals/job_utils/summoner.lua

```diff
@@ -175,6 +175,16 @@ xi.job_utils.summoner.canUseBloodPact = function(player, pet, target, petAbility
             return xi.msg.basic.TARG_OUT_OF_RANGE, 0
         end
 
+        -- check if avatar is asleep
+        if pet:hasStatusEffect(xi.effect.SLEEP_I) or pet:hasStatusEffect(xi.effect.SLEEP_II) then
```

Comment:
Lullaby, pretrify and surely other status effects also impede action.

Reply:
Big should. Test that beofre pushing changes

Reply:
Good call Mow that works for Petrify, both Sleeps, Lullaby, and Stun. There's also a separate issue that was noted where a mob moves too far out of range as the BP is being activated which results in the BP going on cooldown and it eating MP. Will keep that separate from this but wanted to bring it up in case anybody had ideas on a good fix.

Reply:
> There's also a separate issue that was noted where a mob moves too far out of range as the BP is being activated which results in the BP going on cooldown and it eating MP. Will keep that separate from this but wanted to bring it up in case anybody had ideas on a good fix.

Depending on the age of the info, and the avatar, that may be different now. All avatars are converted now so it would be worth double checking and comparing with retail

Reply:
Don't known if I'll catch it later, so commenting now while it is on the mind:

Will be important to track when retail processes the ability/mp consumption now for the out of range issue.  Seem to recall it being consumed regardless in retail back in the day... like the consumption/cooldown was bound to the player action and not the avatar's response to it.  Same thing could happen if someone killed the claimed target before the avatar completed the last BP called (even with wards)--the animation and all couid get interrupted but the timer was already set and MP taken. Wasn't like casting a spell where it waits until it hits the completion point .. fired off when the player called the ability and then the avatar was sort of a seperate action on its own after that?